src/map/status.c

```diff
@@ -3320,220 +3320,355 @@ static int status_calc_homunculus_(struct homun_data *hd, enum e_status_calc_opt
 	return 1;
 }
 
-//Calculates base regen values.
-static void status_calc_regen(struct block_list *bl, struct status_data *st, struct regen_data *regen)
+static void status_calc_regen_pc(struct map_session_data *sd, struct status_data *st, struct regen_data *regen)
 {
-	struct map_session_data *sd;
-	int val, skill_lv, reg_flag;
-	nullpo_retv(bl);
+	nullpo_retv(sd);
 	nullpo_retv(st);
+	nullpo_retv(regen);
+	nullpo_retv(regen->sitting);
+	nullpo_retv(regen->skill);
 
-	if( !(bl->type&BL_REGEN) || !regen )
-		return;
+	struct status_change *sc = &sd->sc;
+	int skill_lv;
```

Comment:
`skill_lv` is first used ~50 lines below in `if ((skill_lv = pc->checkskill(sd, HP_MEDITATIO)) > 0) {` so it should be declared there, I think.